src/test/java/AddUserTest.java

```diff
@@ -0,0 +1,98 @@
+    }
+
+    // Scenario #1:
+    @Test
+    void addUserTest() throws IOException {
+        String zipCode = zipCodeClient.getNewZipCode();
+        User user = User.builder()
+                .name(faker.name().firstName())
+                .age(faker.number().numberBetween(1, 100))
+                .sex(faker.demographic().sex().toUpperCase())
+                .zipCode(zipCode)
+                .build();
+        int postUserStatusCode = userClient.addToUsersList(user);
+        Response<List<User>> getUsersResponse = userClient.getUsersList();
+        Response<List<String>> getZipCodesResponse = zipCodeClient.getZipCodesList();
+
+        Assertions.assertAll(
+                () -> Assertions.assertEquals(CREATED_STATUS, postUserStatusCode),
+                () -> Assertions.assertTrue(getUsersResponse.getBody().contains(user)),
+                () -> Assertions.assertFalse(getZipCodesResponse.getBody().contains(zipCode))
+        );
+    }
+
+    // Scenario #2:
+    @Test
+    void addRequiredUserTest() throws IOException {
+        User user = User.builder()
+                .name(faker.name().firstName())
+                .sex(faker.demographic().sex().toUpperCase())
+                .build();
+        int postUserStatusCode = userClient.addToUsersList(user);
+        Response<List<User>> getUsersResponse = userClient.getUsersList();
+
+        Assertions.assertAll(
+                () -> Assertions.assertEquals(CREATED_STATUS, postUserStatusCode),
+                () -> Assertions.assertTrue(getUsersResponse.getBody().contains(user))
+        );
+    }
+
+    // Scenario #3:
+    @Test
+    void addUnavailableUserTest() throws IOException {
+        User user = User.builder()
+                .name(faker.name().firstName())
+                .age(faker.number().numberBetween(1, 100))
+                .sex(faker.demographic().sex().toUpperCase())
+                .zipCode(faker.number().digits(5))
+                .build();
+        int postUserStatusCode = userClient.addToUsersList(user);
+        Response<List<User>> getUsersResponse = userClient.getUsersList();
+
+        Assertions.assertAll(
+                () -> Assertions.assertEquals(DEPENDENCY_STATUS, postUserStatusCode),
+                () -> Assertions.assertFalse(getUsersResponse.getBody().contains(user))
+        );
+    }
+
+    // Scenario #4:
+    @Test
+    void addDuplicateUserTest() throws IOException {
+        Response<List<User>> getUsersResponse = userClient.getUsersList();
+        List<User> usersList = getUsersResponse.getBody();
+        User user = User.builder()
+                .name(usersList.get(0).getName())
+                .sex(usersList.get(0).getSex())
+                .build();
+        int postUserStatusCode = userClient.addToUsersList(user);
+
+        Assertions.assertAll(
+                () -> Assertions.assertEquals(BAD_STATUS, postUserStatusCode),
+                () -> Assertions.assertFalse(usersList.contains(user))
```

Comment:
you first get a list of users, then you try to save a duplicate user. Then you check whether it is saved in the database based on the list taken earlier. In this case, even if there is a bug, you won't notice it